src/Core/Mesh/TopologicalTriMesh/TopologicalMesh.hpp

```diff
@@ -1,35 +1,130 @@
 #ifndef TOPOLOGICALMESH_H
 #define TOPOLOGICALMESH_H
 
-#include <Core/RaCore.hpp>
 #include <Core/Math/LinearAlgebra.hpp>
+#include <Core/RaCore.hpp>
+#include <Core/Index/Index.hpp>
+
+#include <OpenMesh/Core/Mesh/PolyMesh_ArrayKernelT.hh>
 #include <OpenMesh/Core/Mesh/TriMesh_ArrayKernelT.hh>
 #include <OpenMesh/Core/Utils/Property.hh>
-
 #include <OpenMesh/Core/Mesh/Traits.hh>
 
-namespace Ra {
-namespace Core {
+namespace Ra
+{
+namespace Core
+{
+
+///\class TopoVector3 : small extension to Vector3 for OpenMesh compatibility
+class TopoVector3 : public Ra::Core::Vector3
```

Comment:
If we need this, we also might need implicit cast operators (in both direction) with `Ra::Core::Vector3`